dask/dataframe/core.py

```diff
@@ -2877,7 +2877,9 @@ def values(self):
         Operations that depend on shape information, like slicing or reshaping,
         will not work.
         """
-        return self.map_partitions(methods.values)
+        a = self.map_partitions(methods.values)
+        a._meta = method.values(self._meta)
```

Comment:
This feels like patching a symptom of either `map_partitions` or `methods.values` instead of fixing the root cause. My guess is there's other situations in using `map_partitions` that run into the same issue.

Reply:
Agreed unfortunately I don't see a good way to fix that. If others have suggestions, that would be welcome 🙂

FWICT other things like `to_dask_array` just go through `.values`. So this should fix the common cases that people run into

Reply:
Ah - Looks like I was overthinking it.

I **think** you just need to add `meta=meta` [on this line] (https://github.com/dask/dask/blob/2613608da99b6a46333c0e2ff2000cf20d41faa5/dask/dataframe/core.py#L6692), and the numpy/cupy issue is resolved.